assets/js/googlesitekit/components-gm2/Button/MaybeTooltip.js

```diff
@@ -0,0 +1,72 @@
+/**
+ * MaybeTooltip component.
+ *
+ * Site Kit by Google, Copyright 2025 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+
+/**
+ * Internal dependencies
+ */
+import Tooltip from '../Tooltip';
+
+export default function MaybeTooltip( {
+	children,
+	disabled,
+	tooltip,
+	tooltipTitle,
+	hasIconOnly,
+	tooltipEnterDelayInMS,
+} ) {
+	if (
+		! disabled &&
+		( ( tooltip && tooltipTitle ) || ( hasIconOnly && tooltipTitle ) )
+	) {
+		return (
+			<Tooltip
+				title={ tooltipTitle }
+				enterDelay={ tooltipEnterDelayInMS }
+			>
+				{ children }
+			</Tooltip>
+		);
+	}
+
+	return children;
+}
+
+MaybeTooltip.propTypes = {
+	children: PropTypes.node.isRequired,
+	disabled: PropTypes.bool,
+	tooltip: PropTypes.bool,
+	tooltipTitle: PropTypes.oneOfType( [
+		PropTypes.string,
+		PropTypes.element,
+	] ),
+	hasIconOnly: PropTypes.bool,
+	tooltipEnterDelayInMS: PropTypes.number,
+};
+
+MaybeTooltip.defaultProps = {
+	disabled: false,
+	tooltip: false,
+	tooltipTitle: null,
+	hasIconOnly: false,
+	tooltipEnterDelayInMS: 100,
+};
```

Comment:
Let's use default parameters in the exported function, as `defaultProps` on functional components will eventually be deprecated (see [RFC](https://github.com/reactjs/rfcs/pull/107)).

With that said, as we're already refactoring, we should apply this refactor to the parent component too, if capacity permits, thank you!